### PR TITLE
LR-FHSS API additions

### LINKED ISSUE #4348: LR-FHSS Gateway entity support
#### Summary

The `Gateway` entity should have a field that signals that it supports LR-FHSS

#### Why do we need this?

In order to mark gateways that support features such as LR-FHSS. This allows us to avoid introducing separate frequency plans for each capability combination.

#### What is already there? What do you see now?

No API for this.

#### What is missing? What do you want to see?

A new field.

#### Environment

`v3.14`

#### How do you propose to implement this?

Add the required API fields.

#### Can you do this yourself and submit a Pull Request?

Yes.

### PR DESCRIPTION
**Pull request by adriansmares**
#### Summary

Closes https://github.com/TheThingsNetwork/lorawan-stack/issues/4348
References https://github.com/TheThingsNetwork/lorawan-stack/issues/3806

#### Changes

- Add `supports_lrfhss` field to the `Gateway` entity, and add support for storage in the IS
- Add the `LRFHSS` modulation

#### Testing

Unit testing.

##### Regressions

API additions shouldn't result in regressions.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: api/gateway.proto [hunk 1]
```diff
@@ -161,6 +161,9 @@ message Gateway {
   // Require an authenticated gateway connection. This prevents the gateway from using the UDP protocol and requires authentication when using other protocols.
   bool require_authenticated_connection = 27;
 
+  // The gateway supports the LR-FHSS uplink channels.
+  bool supports_lrfhss = 28;
+
   // next: 28
```
Comment (johanstokking) [inline api/gateway.proto]:
```suggestion
  // next: 29
```

**Review by htdvisser — COMMENTED** [review]
This needs a CHANGELOG entry, other than that, LGTM

**Review by johanstokking — COMMENTED** [review]
Only looked at API

I think it's good that we start declaring gateway capabilities in `Gateway`, just like we have with `EndDevice`. That's going to be useful when filtering frequency plans, configuring the gateway and mapping the gateway. We know little about LR-FHSS, so maybe we get downlink support later, maybe we get a "V2" version. Should we put this under a `LRFHSS` message in `Gateway`, with now only 1 field `supported`, so we don't end up with a bunch of first level fields to give meaning to `supports_lrfhss`?

Also, GS and GCS may need this for configuring gateways (i.e. generating and serving frequency plans). Therefore, for unauthenticated gateways, GS needs to get this field with cluster auth. So I think it should be set in `PublicSafe()`.

**Review by kschiffer — APPROVED** [review]
Codeowned LGTM

**Review by johanstokking — APPROVED** [review]

**Pull request merged**